parley/src/tests/test_wrap.rs

```diff
@@ -0,0 +1,273 @@
+        120.0,
+    );
+}
+
+#[test]
+fn word_break_break_all_everywhere() {
+    let mut env = testenv!();
+
+    test_wrap(
+        &mut env,
+        Some("Most words are short. But Antidisestablishmentarianism is long and needs to wrap."),
+        StyleProperty::WordBreak(WordBreak::BreakAll),
+        ColorBrush::new(0, 128, 0, 255),
+        120.0,
+    );
+}
+
+#[test]
+fn word_break_break_all_min_content_width() {
+    let mut env = testenv!();
+
+    let text = "Hello world!\nLonger line with a looooooooong word.";
+    let mut builder = env.ranged_builder(text);
+    builder.push_default(StyleProperty::WordBreak(WordBreak::BreakAll));
+
+    let mut layout = builder.build(text);
+
+    layout.break_all_lines(Some(layout.min_content_width()));
+    layout.align(None, Alignment::Start, AlignmentOptions::default());
+    // This snapshot will have slightly different line wrapping than the corresponding overflow-wrap test. This is to be
+    // expected and matches browser/CSS behavior.
+    env.check_layout_snapshot(&layout);
+}
+
+#[test]
+fn word_break_wpt007() {
+    // See http://wpt.live/css/css-text/word-break/word-break-break-all-inline-007.tentative.html
+    //
+    // All browsers fail this currently, but we pass it. This means that word_break_break_all_first_half doesn't match
+    // what any browsers do currently, but should be theoretically correct.
+    let mut env = testenv!();
+
+    test_wrap_with_custom_text(
+        &mut env,
+        "aaaaaaabbbbbbbcccccc",
+        Some("bbbbbbb"),
+        StyleProperty::WordBreak(WordBreak::BreakAll),
+        ColorBrush::new(0, 128, 0, 255),
+        55.0,
+    );
+}
+
+#[test]
+fn word_break_keep_all() {
+    let mut env = testenv!();
+
+    let mut test_text = |text, name, wrap_width| {
+        let mut builder = env.ranged_builder(text);
+        builder.push_default(StyleProperty::WordBreak(WordBreak::KeepAll));
+
+        let mut layout = builder.build(text);
+
+        layout.break_all_lines(Some(wrap_width));
+        layout.align(None, Alignment::Start, AlignmentOptions::default());
+        env.with_name(name).check_layout_snapshot(&layout);
+    };
+
+    // These are the word-break-keep-all tests from WPT:
+    // https://wpt.fyi/results/css/css-text/word-break?label=experimental&label=master&aligned
+    test_text("Latin latin latin latin", "latin", 120.0);
+    // These will all show up as boxes because CJK fonts are quite large (several megabytes per language) and could
+    // bloat the repository. Line break analysis should work the same regardless of font, however.
+    test_text("日本語 日本語 日本語", "japanese", 60.0);
+    test_text("한글이 한글이 한글이", "korean", 60.0);
+    // TODO: we fail this test; so does Safari
+    // https://wpt.fyi/results/css/css-text/word-break/word-break-keep-all-003.html
+    // test_text("และ และและ", "thai", 65.0);
+    test_text("フォ フォ", "ID_and_CJ", 30.0);
+    test_text("애기판다 애기판다", "korean_hangul_jamos", 90.0);
```

Comment:
Might be good to add a little comment above this string to clarify it is decomposed, some people may not be familiar with jamo decomposition, and may accidentally normalize this string.

Reply:
I've added a comment about it.